go/analyzer/staticcheck/README.md

```diff
@@ -0,0 +1,57 @@
+# Staticcheck nogo Analyzers
+
+This package is provided as an opt-in part of rules_go.
+
+Projects who have the need to use staticcheck analyzers as part of rules_go's nogo framework may use this package to enable the checks over existing code base
+
+## How to use
+
+1. Please refer to how to use **nogo** in https://github.com/bazelbuild/rules_go/blob/master/go/nogo.rst
+
+
+2. Enable staticcheck dependencies in your WORKSPACE.
+
+```
+load("@io_bazel_rules_go//go/analyzer/staticcheck:deps.bzl", "staticcheck_deps")
+
+staticcheck_deps()
+```
+
+Because this make uses of `go_repository` rules of Gazelle, you should only load it after `gazelle_dependencies`.
```

Comment:
Is this true? I think I saw some documentation from Gazelle that they would like dependencies to be declared after.

Reply:
I think it was true for an earlier iteration on my local laptop, but not true for current version in this PR.
In particular, we can switch the test setup around like this:

```
--- a/tests/core/nogo/staticcheck/staticcheck_test.go
+++ b/tests/core/nogo/staticcheck/staticcheck_test.go
@@ -73,13 +73,13 @@ func alwaysTrue(a int) string {
 }
 `,
                WorkspaceSuffix: `
-load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies")
-
-gazelle_dependencies()
-
 load("@io_bazel_rules_go//go/analyzer/staticcheck:deps.bzl", "staticcheck_deps")

 staticcheck_deps()
+
+load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies")
+
+gazelle_dependencies()
```

And things still work for  `bazel test //tests/core/nogo/staticcheck/...`  🤔

I am not quite sure what was the issue I ran into during development but I cannot reproduce it now so I would be more than happy to drop this sentence in the doc.

Reply:
Further testing showed that the order here does matter as loading gazelle deps first would load a diff version of staticcheck